dotnet/src/Connectors/Connectors.Memory.Weaviate/WeaviateVectorStoreRecordCollectionQueryBuilder.cs

```diff
@@ -68,6 +68,58 @@ public static string BuildSearchQuery<TRecord, TVector>(
         """;
     }
 
+    /// <summary>
+    /// Builds Weaviate search query.
+    /// More information here: <see href="https://weaviate.io/developers/weaviate/api/graphql/get"/>.
+    /// </summary>
+    public static string BuildQuery<TRecord>(
+        Expression<Func<TRecord, bool>> filter,
+        int top,
+        FilterOptions<TRecord> queryOptions,
+        IEnumerable<KeyValuePair<VectorStoreRecordPropertyModel, bool>> orderByProperties,
+        string collectionName,
+        VectorStoreRecordModel model)
+    {
+        var vectorsQuery = queryOptions.IncludeVectors ?
+            $"vectors {{ {string.Join(" ", model.VectorProperties.Select(p => p.StorageName))} }}" :
+            string.Empty;
+
+        string sortPaths = string.Join(",", orderByProperties.Select(property =>
+        {
+            string? sortPath = property.Key switch
+            {
+                VectorStoreRecordKeyPropertyModel key => "id",
+                _ => property.Key.StorageName,
+            };
+
+            return $$"""{ path: ["{{sortPath}}"], order: {{(property.Value ? "asc" : "desc")}} }""";
+        }));
+
+        var sort = string.IsNullOrEmpty(sortPaths) ? string.Empty : $"sort: [ {sortPaths} ]";
+
+        var translatedFilter = new WeaviateFilterTranslator().Translate(filter, model);
+
+        return $$"""
+        {
+          Get {
+            {{collectionName}} (
+              limit: {{top}}
+              offset: {{queryOptions.Skip}}
+              where: {{translatedFilter}}
+              {{sort}}
+            ) {
+              {{string.Join(" ", model.DataProperties.Select(p => p.StorageName))}}
```

Comment:
Is there a possibility to extract common code with the vector-searching variant just above? In many (most?) cases, it seems like the non-vector and vector searching use very similar calls, it would be great to have a single method here called by both with only the differences passed as a parameter etc.

(otherwise I see maintenance headaches ahead because of the duplication... this is in general for the other connectors too...)

Reply:
Ah I see you're discussing this in https://github.com/microsoft/semantic-kernel/pull/11112/files#r2028878158. I'm OK with doing cleanup in a later PR, though at least in this case it seems pretty natural/easy (maybe it's less easy in other cases).